### DISCUSSION TITLE
I can't send message with the wrapped Websocket Object

### DISCUSSION THREAD

**Discussion by alejandrozf** [bigskysoftware/htmx#1840] [Q&A] [answered]
Hi guys

I'm trying to use the wrapped Websocket object with the websocket extension when connection is made, I mean inside the event _htmx:wsOpen_ but it is not working. I tried with _send_ and with _sendInmediatly_ but neither worked.

I can send data by using input in forms but not using the wrapped websocket object

This is the code I'm using:

```
             document.body.addEventListener('htmx:wsOpen', function(evt) {
               evt.detail.socketWrapper.send("some data 1");
               evt.detail.socketWrapper.sendImmediately("some data 2");
             });
```

Could you give me a hand with this?

**Comment by matiboy** [comment, marked as answer]
Hi @alejandrozf looking at the source code, it looks like you must pass a "sendingElt" argument to send and `sendImmediately` otherwise they don't do anything.

So

```js
evt.detail.socketWrapper.sendImmediately("some data 2", document.body);
``` 

should do it

**Reply by alejandrozf** [reply]
Thank you @matiboy !

It worked!
